// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: list_timers.proto

package evaluatord

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"strings"
	"time"
	"unicode/utf8"

	"github.com/golang/protobuf/ptypes"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = ptypes.DynamicAny{}
)

// define the regex for a UUID once up-front
var _list_timers_uuidPattern = regexp.MustCompile("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$")

// Validate checks the field values on ListTimersRequest with the rules defined
// in the proto definition for this message. If any rules are violated, an
// error is returned.
func (m *ListTimersRequest) Validate() error {
	if m == nil {
		return nil
	}

	if m.GetTimer() == nil {
		return ListTimersRequestValidationError{
			field:  "Timer",
			reason: "value is required",
		}
	}

	if v, ok := interface{}(m.GetTimer()).(interface{ Validate() error }); ok {
		if err := v.Validate(); err != nil {
			return ListTimersRequestValidationError{
				field:  "Timer",
				reason: "embedded message failed validation",
				cause:  err,
			}
		}
	}

	return nil
}

// ListTimersRequestValidationError is the validation error returned by
// ListTimersRequest.Validate if the designated constraints aren't met.
type ListTimersRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ListTimersRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ListTimersRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ListTimersRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ListTimersRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ListTimersRequestValidationError) ErrorName() string {
	return "ListTimersRequestValidationError"
}

// Error satisfies the builtin error interface
func (e ListTimersRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sListTimersRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ListTimersRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ListTimersRequestValidationError{}

// Validate checks the field values on ListTimersResponse with the rules
// defined in the proto definition for this message. If any rules are
// violated, an error is returned.
func (m *ListTimersResponse) Validate() error {
	if m == nil {
		return nil
	}

	for idx, item := range m.GetTimers() {
		_, _ = idx, item

		if v, ok := interface{}(item).(interface{ Validate() error }); ok {
			if err := v.Validate(); err != nil {
				return ListTimersResponseValidationError{
					field:  fmt.Sprintf("Timers[%v]", idx),
					reason: "embedded message failed validation",
					cause:  err,
				}
			}
		}

	}

	return nil
}

// ListTimersResponseValidationError is the validation error returned by
// ListTimersResponse.Validate if the designated constraints aren't met.
type ListTimersResponseValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e ListTimersResponseValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e ListTimersResponseValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e ListTimersResponseValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e ListTimersResponseValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e ListTimersResponseValidationError) ErrorName() string {
	return "ListTimersResponseValidationError"
}

// Error satisfies the builtin error interface
func (e ListTimersResponseValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sListTimersResponse.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = ListTimersResponseValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = ListTimersResponseValidationError{}
